src/plugins/opensearch_dashboards_utils/common/state_containers/create_state_container.ts

```diff
@@ -47,12 +46,22 @@ const isProduction =
     ? process.env.NODE_ENV === 'production'
     : !process.env.NODE_ENV || process.env.NODE_ENV === 'production';
 
+const deepFreeze = (value: any) => {
```

Comment:
We have a deepFreeze definition in this utility package, packages/osd-std/src/deep_freeze.ts:
```
import { RecursiveReadonly } from '@osd/utility-types';

/** @public */
export type Freezable = { [k: string]: any } | any[];

/**
 * Apply Object.freeze to a value recursively and convert the return type to
 * Readonly variant recursively
 *
 * @public
 */
export function deepFreeze<T extends Freezable>(object: T) {
  // for any properties that reference an object, makes sure that object is
  // recursively frozen as well
  for (const value of Object.values(object)) {
    if (value !== null && typeof value === 'object') {
      deepFreeze(value);
    }
  }
  return Object.freeze(object) as RecursiveReadonly<T>;
}
```
Could we move this one in in file as well? The difference I could see is that this one  includes function freezing. So maybe we modify deepFreeze in this file as a basic freeze function. Then extend it to include function freezing? Therefore, other files can use it and code is cleaner. What do you think?

Reply:
I agree with Anan. We should port yours to replace the one linked above as yours is better on a bunch of levels. Then import `deepFreeze` from there.